### PR TITLE
fix: so email and phone are show on subscriber in trigger example

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What kind of change does this PR introduce? 
- [x] Bug
- [ ] Feature
- [ ] Docs
- [ ] Other(s)

### Why was this change needed?
https://linear.app/novu/issue/NV-1113/subscriber-variables-doesn-not-exist-in-trigger

### Other information (Screenshots)
Screenshot 2022-10-31 at 14 06 00

### CODE REVIEW

File: apps/web/src/components/activity/ExecutionDetailTrigger.tsx [hunk 1]
```diff
@@ -11,16 +11,34 @@ const TriggerTitle = styled(Text)`
...
-  subscriberVariables: [{ name: 'subscriberId', value: subscriberId }],
-});
-
-export const ExecutionDetailTrigger = ({ step }) => {
-  const { identifier, payload, _subscriberId } = step || {};
-  const trigger = buildTrigger(identifier, _subscriberId, payload);
+const buildTrigger = (identifier, subscriber, payload): INotificationTrigger => {
+  const subscriberVariables = [{ name: 'subscriberId', value: subscriber._id }];
+
+  if (subscriber.email) {
```
Comment (ainouzgali) [inline apps/web/src/components/activity/ExecutionDetailTrigger.tsx:17]: So we should add here for firstName, lastName, and maybe even avatar. As they are all variables that can be used in the template for a subscriber
What do you think?

**Review by ainouzgali — APPROVED** [review]
🤗

File: apps/web/src/components/activity/ExecutionDetailTrigger.tsx (hunk 1 above)
Comment (ainouzgali) [reply apps/web/src/components/activity/ExecutionDetailTrigger.tsx:17]: Oh wait. But we only want to show in the trigger the subscriberVariables that were actually sent - so we keep the data for debugging accurate. Also, if subscriber doesn't have a phone but a phone is needed and sent in the to payload - that's a situation that will also want to be shown. Or if he sent in the trigger a different phone.

Comment (p-fernandez) [reply apps/web/src/components/activity/ExecutionDetailTrigger.tsx:17]: @ainouzgali, does the query brings from DB subscriber variables with information that might be not used in the initial call?

Comment (p-fernandez) [reply apps/web/src/components/activity/ExecutionDetailTrigger.tsx:17]: Yes, you understood me what I mean. But then that means we have no way to exactly know all the subscriber variables used on the trigger. I can't think how we could identify them.

Comment (ainouzgali) [reply apps/web/src/components/activity/ExecutionDetailTrigger.tsx:17]: We do save the subscriberVariables inside the trigger in the template - so we kind of have a way to get those. 
But about the values, I'm not completely sure where they would be if not from the subscriber.

File: apps/web/src/components/activity/ExecutionDetailTrigger.tsx [hunk 2]
```diff
@@ -11,34 +11,25 @@ const TriggerTitle = styled(Text)`
...
-  }
-
-  if (subscriber.phone) {
-    subscriberVariables.push({
-      name: 'phone',
-      value: subscriber.phone,
-    });
-  }
+const buildTrigger = (identifier, subscriberVariables, payload): INotificationTrigger => {
+  subscriberVariables = JSON.parse(subscriberVariables || '{}');
```
Comment (p-fernandez) [inline apps/web/src/components/activity/ExecutionDetailTrigger.tsx:15]: What's the reason behind doing the variable global?

Comment (davidsoderberg) [reply apps/web/src/components/activity/ExecutionDetailTrigger.tsx:15]: global?

Comment (p-fernandez) [reply apps/web/src/components/activity/ExecutionDetailTrigger.tsx:15]: Didn't see it in the arguments, sorry.
Not big fan of modifying them through the memory reference through. Good to go anyway.

File: apps/web/src/components/activity/ExecutionDetailTrigger.tsx [hunk 3]
```diff
@@ -11,16 +11,23 @@ const TriggerTitle = styled(Text)`
...
+  return {
+    type: TriggerTypeEnum.EVENT,
+    identifier,
+    variables: payload ? Object.entries(payload).map(([name, value]) => ({ name, value })) : [],
+    subscriberVariables: Object.keys(subscriberVariables).map((key) => {
+      return {
+        name: key,
+        value: subscriberVariables[key],
+      };
+    }),
```
Comment (p-fernandez) [inline apps/web/src/components/activity/ExecutionDetailTrigger.tsx:24]: You could also use `Object.entries`. Difference would be golfing.

**Review by p-fernandez — APPROVED** [review]
🔥

**Pull request merged**